src/lib/deskflow/ProtocolTypes.h

```diff
@@ -11,365 +11,1046 @@
-// client should reply with a kMsgDInfo.
-extern const char *const kMsgQInfo;
-
-//
-// error codes
-//
-
-// incompatible versions:  primary -> secondary
-// $1 = major version of primary, $2 = minor version of primary.
-extern const char *const kMsgEIncompatible;
-
-// name provided when connecting is already in use:  primary -> secondary
-extern const char *const kMsgEBusy;
-
-// unknown client:  primary -> secondary
-// name provided when connecting is not in primary's screen
-// configuration map.
-extern const char *const kMsgEUnknown;
-
-// protocol violation:  primary -> secondary
-// primary should disconnect after sending this message.
-extern const char *const kMsgEBad;
-
-//
-// structures
-//
-
-//! Screen information
-/*!
-This class contains information about a screen.
-*/
-class ClientInfo
-{
-public:
-  //! Screen position
-  /*!
-  The position of the upper-left corner of the screen.  This is
-  typically 0,0.
-  */
-  int32_t m_x;
-  int32_t m_y;
-
-  //! Screen size
-  /*!
-  The size of the screen in pixels.
-  */
-  int32_t m_w;
-  int32_t m_h;
+/**
+ * @defgroup protocol_files File Transfer Messages
+ * @brief File transfer and drag-and-drop messages (v1.5+)
+ * @{
+ */
 
-  //! Obsolete (jump zone size)
-  int32_t obsolete1;
+/**
+ * @brief File transfer data
+ *
+ * **Message Code**: `"DFTR"`
+ * **Direction**: Primary ↔ Secondary
+ * **Format**: `"DFTR%1i%s"`
+ * **Parameters**:
+ * - `$1`: Transfer mark (1 byte) - Transfer state
+ * - `$2`: Data (string) - Content depends on mark
+ *
+ * **Transfer Marks**:
+ * - `1` (kDataStart): Data contains file size (8 bytes)
+ * - `2` (kDataChunk): Data contains file content chunk
+ * - `3` (kDataEnd): Transfer complete (data may be empty)
+ *
+ * **Example Transfer Sequence**:
+ * ```
+ * "DFTR\x01\x00\x00\x00\x00\x00\x00\x10\x00"  // Start: 4096 bytes
+ * "DFTR\x02[1024 bytes of file data]"          // Chunk: data
+ * "DFTR\x02[1024 bytes of file data]"          // Chunk: data
+ * "DFTR\x02[1024 bytes of file data]"          // Chunk: data
+ * "DFTR\x02[1024 bytes of file data]"          // Chunk: data
+ * "DFTR\x03"                                   // End: complete
+ * ```
+ *
+ * **Protocol Flow**:
+ * 1. Sender initiates with kDataStart containing total file size
+ * 2. Sender sends multiple kDataChunk messages with file content
+ * 3. Sender concludes with kDataEnd to signal completion
+ * 4. Receiver can abort by closing connection
+ *
+ * @see kMsgDDragInfo, EDataTransfer
+ * @since Protocol version 1.5
+ */
+extern const char *const kMsgDFileTransfer;
 
-  //! Mouse position
-  /*!
-  The current location of the mouse cursor.
-  */
-  int32_t m_mx;
-  int32_t m_my;
-};
+/**
+ * @brief Drag and drop information
+ *
+ * **Message Code**: `"DDRG"`
+ * **Direction**: Primary ↔ Secondary
+ * **Format**: `"DDRG%2i%s"`
+ * **Parameters**:
+ * - `$1`: Number of files (2 bytes)
+ * - `$2`: File paths (string) - Null-separated file paths
+ *
+ * **Example**:
+ * ```
```

Comment:
This example/comment appears to end prematurely.

Also, this is deprecated too as it's drag and drop.

```suggestion
 * **Example**:
 * ```
 * TODO
 * ```
 * @deprecated File drag and drop is no longer implemented.
```

Reply:
added depricated